addons/l10n_in/models/account_invoice.py

```diff
@@ -101,7 +101,7 @@ def _post(self, soft=True):
         posted = super()._post(soft)
         gst_treatment_name_mapping = {k: v for k, v in
                              self._fields['l10n_in_gst_treatment']._description_selection(self.env)}
-        for move in posted.filtered(lambda m: m.country_code == 'IN'):
+        for move in posted.filtered(lambda m: m.company_id.account_fiscal_country_id.code == 'IN' and m.is_sale_document()):
```

Comment:
As pointed by @jco-odoo, `country_code` on the account.move is a related of `company_id.account_fiscal_country_id.code` (see the field definition).
So my first comment was wrong, you can just do:
```suggestion
        for move in posted.filtered(lambda m: m.country_code == 'IN' and m.is_sale_document()):
```
Sorry for the back and forths